public/app/features/dashboard-scene/conditional-rendering/ConditionalRenderingGroup.tsx

```diff
@@ -47,6 +53,24 @@ export class ConditionalRenderingGroup extends ConditionalRenderingBase<Conditio
       return true;
     }
 
+    const hasDataRule = !!this.state.value.find((rule) => rule instanceof ConditionalRenderingData);
+
+    if (hasDataRule && this.firstRun) {
+      const item = this.getItem();
+      if (item instanceof AutoGridItem) {
+        const runner = getQueryRunnerFor(item.state.body);
+
+        if (
+          runner &&
+          (!runner.state.data || [LoadingState.NotStarted, LoadingState.Loading].includes(runner.state.data.state))
+        ) {
+          return true;
+        } else {
+          this.firstRun = false;
+        }
+      }
+    }
```

Comment:
Would be nice to have a test for this

Reply:
this felels a bit to hacky a solution. I think a rethink on how hiding works might be needed

Reply:
this felels a bit to hacky a solution. I think a rethink on how hiding works might be needed

Reply:
@Sergej-Vlasov what scenario did the above lines solve? (lines 56-70).

Reply:
But that feels like an easy fix in the data condition, it can check if the data is undefined/(or data provider is in active and return from evaluate in a way so that it's always rendered)